examples/misc/keyword_plotting.py

```diff
@@ -0,0 +1,28 @@
+"""
+======================
+Plotting with keywords
+======================
+
+There are some instances where you have data in a format that lets you
+access particular variables with strings. For example, with
+:class:`numpy.recarray` or :class:`pandas.DataFrame`.
+
+Matplotlib allows you provide such an object with
+the ``data`` keyword argument. If provided, then you may generate plots with
+the strings corresponding to these variables.
+"""
+
+import numpy as np
+import matplotlib.pyplot as plt
+np.random.seed(19680801)
+
+data = {'a': np.arange(50),
+        'c': np.random.randint(0, 50, 50),
+        'd': np.random.randn(50)}
+data['b'] = data['a'] + 10 * np.random.randn(50)
+data['d'] = np.abs(data['d']) * 100
+
+fig, ax = plt.subplots()
+ax.scatter('a', 'b', c='c', s='d', data=data)
+ax.set(xlabel='a', ylabel='b')
```

Comment:
Just to be clear, this doesn't try to somehow pull something out of `data`; maybe use something like 'Entry a'/'Column a' or similar to avoid implying that.